engine/src/silo/mod.rs

```diff
@@ -0,0 +1,196 @@
+use aurora_engine_sdk::io::IO;
+#[cfg(feature = "contract")]
+use aurora_engine_sdk::{env::Env, types::SdkUnwrap};
+use aurora_engine_types::account_id::AccountId;
+use aurora_engine_types::storage::{bytes_to_key, KeyPrefix};
+use aurora_engine_types::types::{Address, Wei};
+use aurora_engine_types::AsBytes;
+
+#[cfg(feature = "contract")]
+use crate::engine::EngineErrorKind;
+use crate::prelude::Vec;
+
+use parameters::{WhitelistArgs, WhitelistKindArgs, WhitelistStatusArgs};
+use whitelist::Whitelist;
+pub use whitelist::WhitelistKind;
+
+pub mod parameters;
+mod whitelist;
+
+const GAS_COST_KEY: &[u8] = b"GAS_COST_KEY";
+
+/// Return fixed gas cost.
+pub fn get_fixed_gas_cost<I: IO>(io: &I) -> Option<Wei> {
+    let key = fixed_gas_cost_key();
+    io.read_u256(&key).ok().map(Wei::new)
+}
+
+/// Set fixed gas cost.
+pub fn set_fixed_gas_cost<I: IO>(io: &mut I, cost: Option<Wei>) {
+    let key = fixed_gas_cost_key();
+
+    if let Some(cost) = cost {
+        io.write_storage(&key, &cost.to_bytes());
+    } else {
+        io.remove_storage(&key);
+    }
+}
+
+/// Add an entry to a white list depending on a kind of list types in provided arguments.
+pub fn add_entity_to_whitelist<I: IO + Copy>(io: &I, args: &WhitelistArgs) {
```

Comment:
@aleksuss should we rename this from add_entity to add_entry? just for consistency